Gems/Archive/Code/Source/Clients/ArchiveTOCView.h

```diff
@@ -0,0 +1,109 @@
+        //! It's length is based on the file count value in the Archive Header Section
+        AZStd::span<ArchiveTocFilePathIndex const> m_filePathIndexTable{};
+
+        //! ArchiveFilePathTable is a view into the start of the blob for file paths
+        using ArchiveFilePathTable = AZStd::string_view;
+        ArchiveFilePathTable m_filePathBlob{};
+
+        //! pointer to block offset table which stores the compressed size of all blocks within the archive
+        AZStd::span<ArchiveBlockLineUnion const> m_blockOffsetTable{};
+    };
+
+    //! Options which allows configuring which sections of the table of contents
+    //! should be validated.
+    //! NOTE: The Block Offset table takes the longest time to validate
+    //! as it verifies the number of block lines in the table of contents
+    //! is equivalent to the number of block lines each file should have
+    //! It does this by calculating the number of block lines a file should
+    //! have by examining each file uncompressed size
+    struct ArchiveTocValidationOptions
+    {
+        bool m_validateFileMetadataTable{ true };
+        bool m_validateFileIndexTable{ true };
+        bool m_validateBlockOffsetTable{ true };
+    };
+    //! Validates the Table of Contents data
+    //! This is a potentially lengthy operation as it verifies
+    //! That the each archived file in the Table of Contents
+    //! has the correct number of blocks based on whether it is
+    //! uncompressed(file is stored contiguously with no entry in the block table)
+    //! or compressed(the number of blocks stored is based on the uncompressed size of the file split into 2 MiB chunks);
+    ArchiveTocValidationResult ValidateTableOfContents(const ArchiveTableOfContentsView& tocView,
+        const ArchiveHeader& archiveHeader,
+        const ArchiveTocValidationOptions& validationOptions = {});
+
+    //! Visitor invoked for each deleted file path index entry
+    //! It gets passed in the offset within the file path blob table where the deleted file path resides
+    //! and the length of that file path
+    //! @param filePathBlobOffset Offset into the raw archive TOC file path blob table where the deleted file path starts
+    //! @param filePathSize length of the deleted file path. The value is guaranteed to be >0
+    using FilePathIndexEntryVisitor = AZStd::function<void(AZ::u32 filePathBlobOffset, AZ::u16 filePathSize)>;
+
+    //! Enumerates each file path index found in the TOC View
+    //! @param callback to invoke when a deleted path index entry is found
+    //! @param tocView view structure overlaying the raw table of contents data
+    //! @return the number of file path index entries visited
+    size_t EnumerateFilePathIndexOffsets(FilePathIndexEntryVisitor callback,
+        const ArchiveTableOfContentsView& tocView);
+} // namespace Archive
+
+// Implementation for any struct functions
+#include "ArchiveTOCView.inl"
```

Comment:
The include here is private and only meant to grab the "ArchiveTOCView.inl" in the current directory.
It should be preferred to not search the list of include paths to avoid using another potential `ArchiveTOCView.inl` file from a different library.

Reply:
Nvm as you pointed out this is internal.